pcs/constraint.py

```diff
@@ -1195,6 +1197,19 @@ def location_rule(lib, argv, modifiers):
             "resource-discovery": None,
         },
     )
+
+    if options["node"] == None or not options["node"]:
+        warn("There's no actual 'location' specified in this constraint, you can specify it by node. ")
```

Comment:
This is a misunderstanding of how location constraints with rules work.

There is no `node` option. The node is specified by the rule itself. For example, rule `node_size gte 42` applies to all nodes which have the value of their attribute `node_size` greater than or equal to `42`.

If, however, there are only date expression in the rule, like shown in #235, then the rule applies to all nodes and is therefore pointless in a location constraint.

You can check `pcs.lib.cib.rule.tools` to see how to analyze a rule to figure out if it contains node expressions. The module, however, is part of pcs library, new architecture, and is not compatible with the rule parser used in location constraints.